aws/resource_aws_nat_gateway.go

```diff
@@ -27,7 +28,7 @@ func resourceAwsNatGateway() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"allocation_id": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```

Comment:
is this optional for both types of connectivity_types, public and private? the new test definitely checks out for the latter 👍

Reply:
The documentation update mentions that it's required for `connectivity_type = "public"`.